src/main/scala/vsys/utils/JsonFileStorage.scala

```diff
@@ -51,34 +52,32 @@ object JsonFileStorage {
   }
 
   def save[T](value: T, path: String, key: Option[SecretKeySpec])(implicit w: Writes[T]): Unit = {
-    var file: Option[PrintWriter] = None
-    try {
-      val folder = new File(path).getParentFile
-      if (!folder.exists())
-        folder.mkdirs()
-      file = Option(new PrintWriter(path))
-      file.foreach {
-        val json = Json.toJson(value).toString()
-        val data = key.fold(json)(k => encrypt(k, json))
-        _.write(data)
+    val folder = new File(path).getParentFile
+    if (!folder.exists())
+      folder.mkdirs()
+    val file: Option[PrintWriter] = Option(new PrintWriter(path))
+    Try {
+      ultimately {
+        file.foreach(_.close())
+      } {
+        file.foreach {
+          val json = Json.toJson(value).toString()
+          val data = key.fold(json)(k => encrypt(k, json))
+          _.write(data)
+        }
       }
-    } finally {
-      file.foreach(_.close())
     }
   }
 
   def save[T](value: T, path: String)(implicit w: Writes[T]): Unit =
     save(value, path, None)
 
   def load[T](path: String, key: Option[SecretKeySpec] = None)(implicit r: Reads[T]): T = {
-    var file: Option[BufferedSource] = None
-    try {
-      file = Option(Source.fromFile(path))
-      val data = file.get.mkString
-      Json.parse(key.fold(data)(k => decrypt(k, data))).as[T]
-    } finally {
-      file.foreach(_.close())
-    }
+    val file: BufferedSource = Source.fromFile(path)
```

Comment:
will aways return a valid BufferedSource?

Reply:
An `FileNotFoundException` could be thrown if `path` not exist.
But in origin code, no catch block implemented. If `Source.fromFile(path)` failed, same exception will be thrown. And the finally block does noting to `None` file option.
So I thought the exception here is fatal, that application cannot continue working

Reply:
ic